api/turing/cluster/controller.go

```diff
@@ -488,7 +489,23 @@ func (c *controller) waitKnativeServiceReady(
 	for {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("timeout waiting for service %s to be ready", svcName)
+			terminationMessage := c.getKnativePodTerminationMessage(svcName, namespace)
+			if terminationMessage == "" {
+				// Pod was not created (as with invalid image names), get most recent status message
+				// from the knative service
+				svc, err := services.Get(svcName, metav1.GetOptions{})
+				if err != nil {
+					terminationMessage = err.Error()
+				} else {
+					failedCondition := svc.Status.GetCondition(knativeapis.ConditionType("ConfigurationsReady"))
+					if failedCondition != nil && failedCondition.IsFalse() {
+						terminationMessage = failedCondition.GetMessage()
+					} else {
+						terminationMessage = "cause unknown"
```

Comment:
In this case, in addition to saying the cause is unknown, can we maybe also provide a bit more context, maybe just listing all the Condition statuses for the Knative service object?

Reply:
Thanks, David. I changed this so that we dump all the status conditions from the service if the pod termination message is not available.